### PR TITLE
Product Component

### PR DESCRIPTION
**Pull request by Chandrika-Rani-Infrrd**
updated:products component with product data containing name and price
updated:products spec file with few test cases
updated:set default path to product component in router module

### CODE REVIEW

File: src/app/products/products.component.spec.ts [hunk 1]
```diff
@@ -22,4 +23,20 @@ describe('ProductsComponent', () => {
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it("does products name match to its price",()=>{
+    const [first,second]=component.products;
```
Comment (lloydaf) [inline src/app/products/products.component.spec.ts]: hmm how does this test make sense?

File: src/app/products/products.component.spec.ts [hunk 2]
```diff
@@ -22,4 +23,20 @@ describe('ProductsComponent', () => {
...
+    expect(second.name).toEqual("college Bag");
+    expect(second.price).toEqual(450); 
+  })
+
+  it("is onclick invoked when click button is clicked",()=>{
+    spyOn(component, 'onClick');
+    let button =  fixture.debugElement.query(By.css('button'));
+    button.triggerEventHandler('click',null);
+    fixture.detectChanges();
+    expect(component.onClick).toHaveBeenCalled();
```
Comment (lloydaf) [inline src/app/products/products.component.spec.ts:32]: 💯

File: src/app/products/products.component.html [hunk 3]
```diff
@@ -1 +1,11 @@
-<p>products works!</p>
+
+<h1 class="products">Products</h1>
+<div class="productsContainer">
+    <ul class="productList" *ngFor="let product of products">
+        <li class="productContent">{{product.name}}</li>
```
Comment (lloydaf) [inline src/app/products/products.component.html]: ideally, product should be it's own component

File: src/app/products/products.component.spec.ts (hunk 1 above)
Comment (Chandrika-Rani-Infrrd) [reply src/app/products/products.component.spec.ts]: different products have different price .if shoes price is 8000 then if i assign any different price value it shows error

Comment (Chandrika-Rani-Infrrd) [reply src/app/products/products.component.spec.ts]: first index should contain the first hard coded name and its value

Comment (lloydaf) [reply src/app/products/products.component.spec.ts]: So you shouldn't write tests for configurations. Configurations can change. Your products can change. It shouldn't matter. Write tests to ensure your code works the way you want it to. :)

**Pull request merged**